lib/amqp/session.rb

```diff
@@ -157,7 +158,7 @@ def initialize(*args, &block)
 
       # EventMachine::Connection's and Adapter's constructors arity
       # make it easier to use *args. MK.
-      @settings                           = Settings.configure(args.first)
+      @settings                           = AMQ::Settings.configure(args.first)
```

Comment:
This completely ignored `other_options` propagated by `AMQP.connect`, which breaks error handling and makes at least one test fail.